imod/mf6/riv.py

```diff
@@ -162,3 +180,225 @@ def _validate(self, schemata, **kwargs):
+
+            regridded_package_data["conductance"] = distribute_riv_conductance(
+                distributing_option_riv,
+                riv_allocation[0],
+                conductance,
+                target_top,
+                target_bottom,
+                conductance,
+                regridded_package_data["stage"],
+                regridded_package_data["bottom_elevation"],
+            )
+
+            # create layered arrays of stage and bottom elevation
+            layered_stage = regridded_package_data["stage"].where(riv_allocation[0])
+            layered_stage = enforce_dim_order(layered_stage)
+            regridded_package_data["stage"] = layered_stage
+
+            layered_bottom_elevation = regridded_package_data["bottom_elevation"].where(
+                riv_allocation[0]
+            )
+            layered_bottom_elevation = enforce_dim_order(layered_bottom_elevation)
+
+            # due to regridding, the layered_bottom_elevation could be smaller than the
+            # bottom, so here we overwrite it with bottom if that's
+            # the case.
+            layered_bottom_elevation = xr.where(
+                target_bottom > layered_bottom_elevation,
+                target_bottom,
+                layered_bottom_elevation,
+            )
+
+            regridded_package_data["bottom_elevation"] = layered_bottom_elevation
+
+        # update the conductance of the river package to account for the infiltration
+        # factor
+        drain_conductance, river_conductance = cls.split_conductance(
+            regridded_package_data["conductance"], infiltration_factor
+        )
+        regridded_package_data["conductance"] = river_conductance
+        regridded_package_data.pop("infiltration_factor")
+
+        river_package = River(**regridded_package_data)
+        # create a drtainage package with the conductance we comptuted from the infiltration factor
+        drainage_package = cls.create_infiltration_factor_drain(
+            regridded_package_data["stage"],
+            drain_conductance,
+        )
+        # remove River package if its mask is False everywhere
+        mask = ~np.isnan(river_conductance)
+        if np.any(mask):
+            river_package = river_package.mask(mask)
+        else:
+            river_package = None
+
+        # remove Drainage package if its mask is False everywhere
+        mask = ~np.isnan(drain_conductance)
+        if np.any(mask):
+            drainage_package = drainage_package.mask(mask)
+        else:
+            drainage_package = None
+        return (river_package, drainage_package)
+
+    @classmethod
+    def create_infiltration_factor_drain(
+        cls,
+        drain_elevation,
+        drain_conductance,
+    ):
+        # create a drainage package from the river package, to account for the infiltration factor.
+        # this factor is optional in imod5, but it does not exist in MF6, so we mimic its effect
+        # with a Drainage boundary.
```

Comment:
Please convert this to a docstring instead of comment.

Reply:
done